### ISSUE TITLE
echo newline:false does not wrap text until resize

### ISSUE THREAD

**Issue by ordo-n** [jcubic/jquery.terminal#952]
### Issue summary

When I use multiple
term.echo('qwerty', {newline:false}');
it correctly doesnt add a newline, but it incorrectly doesnt process newlines until I resize the window

### Expected behavior

Should not have to resize window for newline to appear

### Actual behavior

Trails off to the right

### Steps to reproduce

for(let i = 0; i < 100; i+=1)
term.echo('qwerty', {newline:false}');

### Browser and OS

All browsers

### Additional notes

before resize
![image](https://github.com/jcubic/jquery.terminal/assets/166300150/b65d1277-fbaa-462b-8d63-41589a8318d5)
after resize
![image](https://github.com/jcubic/jquery.terminal/assets/166300150/0c08bbee-2be0-4685-bfdc-d1607d46e3ac)

**Comment by jcubic** [comment]
Thanks for the report, here is repo https://codepen.io/jcubic/pen/wvbmpgX

This is another problem with `newline: false` that was integrated with the library by contributor.

**Comment by jcubic** [comment]
You can fix this with a hack:

```javascript
for(let i = 0; i < 100; i+=1) {
    term.echo('qwerty', { newline: false });
    term.update(-1, ...term.get_output(true).pop());
}
```

This will force having a clean partial render (partial is the output without newline), by refreshing the last line after each echo.

I will probably do something similar inside the library, but without creating garbage from `Terminal::get_output()`. If there are two partials in a row, they need to be processed as one.

**Comment by jcubic** [comment]
It seems this don't solve the problem.

**Comment by jcubic** [comment]
It's a challenging bug, it may take a while to fix. I'm not sure how to handle one edge case.

**Comment by jcubic** [comment]
The bug was fixed and released with version 2.42.1.
